lib/charms/alertmanager_k8s/v0/alertmanager_remote_configuration.py

```diff
@@ -453,3 +470,9 @@ def _load_templates_file(path: str) -> str:
             return templates
         except (FileNotFoundError, OSError, ValueError) as e:
             raise ConfigReadError(path) from e
+
+    def _clear_relation_data(self) -> None:
```

Comment:
Do we really want to clear the relation data? If there was a valid configuration before and it is now invalid, wouldn't it make sense to keep the previous valid configuration?

Reply:
I asked the same question while discussing this with COS team and the reason for clearing the data is that the relation data should reflect the state of the model. Keeping last valid configuration would break that convention. 
From my perspective I can add that it probably makes sense, because if you decided to change the config (regardless of whether your change breaks the config or not) you're probably not interested in using the old config anyway.

Reply:
Ok, I can understand the reasons.

As an old ops guy, I disagree because I would highly prefer to get alarms with the old configs than no alerts at all. Will clearing the relation data impact the alertmanager workload?

Reply:
Yes, if we clear the relation data bag, Alertmanager will switch back to the default config.